### ISSUE TITLE
[Bug] Fields use default value when existing or old value is null

### ISSUE THREAD

**Issue by nbolender** [Laravel-Backpack/CRUD#3678]
# Bug report

### What I did

- Create a field with a default value, but the field is nullable
```php
// addField()
[
    'name' => 'tickets_url',
    'label' => 'Tickets URL',
    'type' => 'url',
    'default' => 'https://example.com',
],

...

// FormRequest
public function rules() {
    return [
        'tickets_url' => 'nullable|url',
    ];
}
```

- Create an object, but erase the default value and leave the field blank. (stored as null in database)
- Open edit form

Alternatively:

- Fill out the create/edit form, but erase the default value and leave the field blank.
- Cause some other validation error such that the form is shown again with 'old' values

### What I expected to happen

tickets_url field value should be blank since its existing value is null

### What happened

tickets_url field value is `https://example.com` (the default value).

This seems to be a fatal flaw in the logic used to get the field value for most fields. That logic is usually:

`value="{{ old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? '' }}"`

The old value and/or the existing value is null, which causes the value to erroneously fallback to the default.

I think this issue is related, but the proposed fix does not solve this case involving the default field value: https://github.com/Laravel-Backpack/CRUD/issues/1816

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
PHP 7.3.27 (cli) (built: Feb 27 2021 18:04:51) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.27, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.27, Copyright (c) 1999-2018, by Zend Technologies

### LARAVEL VERSION:
v7.30.4@9dd38140dc2924daa1a020a3d7a45f9ceff03df3

### BACKPACK VERSION:
4.1.41@44770347b918e42c052d0a138c24535b43e32472

**Comment by pxpm** [comment]
Hello @nbolender hope to find you well. 

Thanks for the report, and sorry for the frustration to have to deal with this.

Can you check if: https://github.com/Laravel-Backpack/CRUD/pull/3054 fixes your problem ? 

Is the solution we have laying around to fix this, but it's a BC, that's why it's on hold. 

Let me know,
Pedro

**Comment by nbolender** [comment]
Hi @pxpm 

I merged that PR against the latest. It solves the problem of the default appearing after a validation error, but the default is still shown on the initial edit form when the stored value is `null`.

This is because `oldOrFallback()` evaluates to `null` (correct), but `$field['value']` is also null, so it falls back to `$field['default']`.

**Comment by pxpm** [comment]
Hello @nbolender 

I'v just updated PR #3054 with current branch and fixed the scenario you describe.

It should be all good now. Do you have desire to give it a try ? 

Note that branch is now even with master, I'v fixed the merging conflicts.

Best,
Pedro

**Comment by nbolender** [comment]
@pxpm Seems to work for me! 🎉

**Comment by tabacitu** [comment]
Let's close this in favour of #3054 , which will be merged soon into the 4.2 branch 🎉

Thanks for opening this @nbolender ... and for the patience 👀 😅
